posts/2021-09-03-microprofile-21009.adoc

```diff
@@ -0,0 +1,268 @@
+
+
+[#bugs]
+== Notable bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A21009+label%3A%22release+bug%22[full list of bugs fixed in 21.0.0.9].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/16700[Improve featureUtility performance with remote repository]
++
+Previously, the link:{url-prefix}/docs/latest/reference/command/command-overview.html#featureUtility[`featureUtility`] would take a long time to resolve features when using a remote repository with a slow network. Artifacts would be fetched one at a time which slowed things down dramatically. This issue was fixed by fetching and downloading multiple artifacts at the same time from Maven repository.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18237[Ignore FFDC for IOExceptions in handleMessage]
++
+A bug was discovered where first failure data capture (FFDC) files were incorrectly produced. This would happen when bad JSON data was passed to REST APIs that are using the `jaxrs-2.1` feature with `BYO Jackson`. This issue could occur when a JAX-RS resource or provider throws any `IOException`, and was fixed by preventing FFDC from logging the `IOException`.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18155[JSF faces-config parser throws NPE when namespace missing]
++
+Previous changes made in order to support Jakarta Faces 3.0 added logic to check for the new `https://jakarta.ee/xml/ns/jakartaee` namespace in `faces-config.xml` files. Because the new logic uses `String.contentEquals()`, if a `faces-config.xml` file declares a version but not a namespace, a `NullPointerException` would be thrown during app startup. This affects all JSF features on Liberty: jsf-2.0, jsf-2.2, jsf-2.3, and faces-3.0.  
+To fix this issue, the logic has now been changed to use `String.equals()`. Please note that while the NullPointerException has been fixed, valid `faces-config.xml` descriptors should declare valid namespaces, ie. `http://java.sun.com/xml/ns/javaee`, `http://xmlns.jcp.org/xml/ns/javaee`, or `https://jakarta.ee/xml/ns/jakartaee`.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18009[Wrong char count in ServletOutputStream with non-ASCII characters skips content]
++
+A bug was discovered where the `ServletOutputStream` would skip characters when using `print` or `println` to print content with multi-byte non-ASCII characters. This bug was fixed by changing the `WCOutputStream` to write the length of the encoded bytes rather than the length of the original string.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18002[@Schema(multipleOf = ) throws NumberFormatException in mpOpenAPI-2.0 feature]
```

Comment:
The text used for this link is wrong (`NumberFormatException` was not involved) and also does not match the title of the issue it links to.

Reply:
I note as well that more than just the mpOpenApi-2.0 feature was affected - though the issue only talks about mpOpenApi-2.0. I'll update the issue to be clear that older versions of mpOpenApi were also affected.